src/pages/fr/core-concepts/endpoints.md

```diff
@@ -0,0 +1,280 @@
+  return {
+    body: buffer,
+    encoding: 'binary',
+  };
+}
+```
+
+Vous pouvez également saisir vos fonctions de points de terminaison en utilisant le type `APIRoute` :
+
+```ts
+import type { APIRoute } from 'astro';
+
+export const get: APIRoute = async function get ({params, request}) {
+...
+```
+
+### `params` et routage dynamique
+
+Les points de terminaison supportent les même fonctionnalités de [routes dynamiques](/fr/core-concepts/routing/#routes-dynamiques) que les pages. Nommez votre fichier avec un nom de paramètre entre crochets et exportez une [fonction `getStaticPaths()`](/fr/reference/api-reference/#getstaticpaths). Ensuite, vous pouvez accéder au paramètre en utilisant la propriété `params` passé à la fonction du point de terminaison :
+
+```ts title="src/pages/[id].json.ts"
+import type { APIRoute } from 'astro';
+
+const usernames = ["Sarah", "Chris", "Dan"]
+
+export const get: APIRoute = ({ params, request }) => {
+  const id = params.id;
+  return {
+    body: JSON.stringify({
+      name: usernames[id]
+    })
+  }
+};
+
+export function getStaticPaths () {
+  return [ 
+    { params: { id: "0"} },
+    { params: { id: "1"} },
+    { params: { id: "2"} },
+  ]
+}
+```
+
+Ceci générera trois points de terminaison JSON au moment de la compilation : `/api/1.json`, `/api/2.json`, `/api/3.json`. Le routage dynamique avec les points de terminaison fonctionne de la même façon qu'avec les pages, mais comme le point de terminaison est une fonction et pas un composant, les [props](/fr/reference/api-reference/#data-passing-with-props) ne sont pas supportées.
+
+### `request`
+
+Tous les points de terminaison reçoivent une propriété `request`, mais en mode statique, vous avez uniquement accès à `request.url`. Ceci retourne l'URL complète du point de terminaison en cours et fonctionne de la même façon que [Astro.request.url](/fr/reference/api-reference/#astrorequest) le fait pour les pages.
+
+```ts title="src/pages/request-path.json.ts"
+import type { APIRoute } from 'astro';
+
+export const get: APIRoute = ({ params, request }) => {
+  return {
+    body: JSON.stringify({
+      path: new URL(request.url).pathname
+    })
+  };
+}
+```
+
+## Points de terminaison Serveur(Routes d'API)
+
+Tout ce qui est décrit dans la section des points de terminaison statiques peut également être utilisé en mode SSR : Les fichiers peuvent exporter une fonction `get` qui reçoit un [objet context](/fr/reference/api-reference/#endpoint-context) avec des propriétés similaires à l'objet `Astro` global.
+
+Mais, à la différence du mode `static`, lorsque vous configurez le mode `server`, les points de terminaison seront construits à la demande. Ceci dévérouillent de nouvelles fonctionnalités qui ne sont pas disponibles au moment de la compilation, et vous permet de construire des routes d'API qui écoutent les demandes et exécutent le code de manière sécurisée sur le serveur au moment de l'exécution.
```

Comment:
Not sure about whether this is a typo or not, I think it should be "deverouille" instead of "deverouillent".